### PR TITLE
Docs: custom block category add icon info

### PR DESCRIPTION
**Pull request by ajitbohra**
With https://github.com/WordPress/gutenberg/pull/10651 custom block categories can have a custom icon.

This PR adds the same to docs https://wordpress.org/gutenberg/handbook/extensibility/extending-blocks/#managing-block-categories

cc: @tyxla

### CODE REVIEW

File: docs/extensibility/extending-blocks.md [hunk 1]
```diff
@@ -333,9 +333,13 @@ function my_plugin_block_categories( $categories, $post ) {
 			array(
 				'slug' => 'my-category',
 				'title' => __( 'My category', 'my-plugin' ),
+				'icon'  => 'screenoptions',
```
Comment (chrisvanpatten) [inline docs/extensibility/extending-blocks.md]: I think it's best if we use an example of an SVG here. Over the long term we want to move away from suggesting Dashicons and push folks to use material icons, and this is a good opportunity to do that.

File: docs/extensibility/extending-blocks.md [hunk 2]
```diff
@@ -333,9 +333,13 @@ function my_plugin_block_categories( $categories, $post ) {
...
 			),
 		)
 	);
 }
 add_filter( 'block_categories', 'my_plugin_block_categories', 10, 2 );
 ```
+
+An icon key/value can be specified to add an icon to block category. These can be any of [WordPress Dashicons](https://developer.wordpress.org/resource/dashicons/), or a custom `svg` element.
```
Comment (chrisvanpatten) [inline docs/extensibility/extending-blocks.md]:
```suggestion
You can also display an icon with your block category by setting an `icon` attribute. The value can be the slug of a [WordPress Dashicon](https://developer.wordpress.org/resource/dashicons/), or a custom `svg` element.
```

**Review by chrisvanpatten — CHANGES_REQUESTED** [review]
Just a few small suggestions :)

File: docs/extensibility/extending-blocks.md (hunk 1 above)
Comment (ajitbohra) [reply docs/extensibility/extending-blocks.md]: I am sure, most of the devs would use custom icons :)

Will update the PR to include SVG or should we both ?

Comment (chrisvanpatten) [reply docs/extensibility/extending-blocks.md]: I think just an SVG example is okay.

Comment (ajitbohra) [reply docs/extensibility/extending-blocks.md]: Updated

File: docs/extensibility/extending-blocks.md [hunk 3]
```diff
@@ -333,7 +333,7 @@ function my_plugin_block_categories( $categories, $post ) {
 			array(
 				'slug' => 'my-category',
 				'title' => __( 'My category', 'my-plugin' ),
-				'icon'  => 'screenoptions',
+				'icon'  => '<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path fill="none" d="M0 0h24v24H0V0z" /><path d="M19 13H5v-2h14v2z" /></svg>',
```
Comment (chrisvanpatten) [inline docs/extensibility/extending-blocks.md]: Small thing, but we also want to encourage block devs to include `width` and `height attributes.

**Review by chrisvanpatten — APPROVED** [review]
Approving but also it'd be great to get width/height attributes in as per my comment.

File: docs/extensibility/extending-blocks.md [hunk 4]
```diff
@@ -333,9 +333,13 @@ function my_plugin_block_categories( $categories, $post ) {
...
 				'title' => __( 'My category', 'my-plugin' ),
+				'icon'  => '<svg viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path fill="none" d="M0 0h24v24H0V0z" /><path d="M19 13H5v-2h14v2z" /></svg>',
 			),
 		)
 	);
 }
 add_filter( 'block_categories', 'my_plugin_block_categories', 10, 2 );
 ```
+
+You can also display an icon with your block category by setting an `icon` attribute. The value can be the slug of a [WordPress Dashicon](https://developer.wordpress.org/resource/dashicons/), or a custom `svg` element.
```
Comment (tyxla) [inline docs/extensibility/extending-blocks.md]: FWIW, components and functions are supported as well - you can refer to https://github.com/wordpress/gutenberg/blob/729fa981795ba1ca90c90eccff6b8e6da38d19f9/packages/components/src/icon/README.md for details, or maybe link to it here in the docs.

File: docs/extensibility/extending-blocks.md (hunk 3 above)
Comment (ajitbohra) [reply docs/extensibility/extending-blocks.md]: Icon component is used to render proper icons and current it enforces default width for dashicon and svg icons. Let me know if should add the width/height to example.

File: docs/extensibility/extending-blocks.md (hunk 4 above)
Comment (ajitbohra) [reply docs/extensibility/extending-blocks.md]: Would be good to link to Icon component for all available options. I guess the same is applicable for block icon also as we are using the same icon component.

File: docs/extensibility/extending-blocks.md (hunk 3 above)
Comment (tofumatt) [reply docs/extensibility/extending-blocks.md]: If there's a default width that works I think this is fine.

File: docs/extensibility/extending-blocks.md (hunk 4 above)
Comment (tofumatt) [reply docs/extensibility/extending-blocks.md]: These are PHP docs though, so referencing a component or function isn't really applicable. I think this README is fine.

**Review by tofumatt — APPROVED** [review]
Works for me!

**Pull request merged**